Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,238 @@
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager: Could not find {symbol} in Portfolio");
+                }
+
+                try
+                {
+                    var consolidator = default(IDataConsolidator);
+                    var security = _portfolio.Securities[symbol];
+                    BaseData lastData = security.Cache.GetData<TradeBar>();
+
+                    // For high resolution data, create a consolidator that will compute
+                    // the AUM Capacity after a 5-minute period bar is closed
+                    foreach (var config in configs.Where(x => x.Resolution < Resolution.Hour))
+                    {
+                        if (config.Type.IsAssignableFrom(typeof(TradeBar)))
+                        {
+                            consolidator = new TradeBarConsolidator(TimeSpan.FromMinutes(5));
+                        }
+
+                        if (config.Type.IsAssignableFrom(typeof(Tick)) &&
+                            config.TickType == TickType.Trade)
+                        {
+                            consolidator = new TickConsolidator(TimeSpan.FromMinutes(5));
+                            lastData = security.Cache.GetData<Tick>();
+                        }
+
+                        if (consolidator == null)
+                        {
+                            continue;
+                        }
+
+                        // Warm up the consolidator to mark the begging of the period
+                        if (lastData != null)
+                        {
+                            consolidator.Update(lastData);
+                        }
+
+                        config.Consolidators.Add(consolidator);
+                        consolidator.DataConsolidated += (s, data) =>  Update(s, data, orderEvent.OrderId);
+                        return;
+                    }
+
+                    // For low resolution, use the last data available
+                    if (lastData != null)
+                    {
+                        Update(sender, lastData, orderEvent.OrderId);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager: {e}");
+                }
+            };
+        }
+
+        /// <summary>
+        /// Updates the AUM Capacity with the latest data available after a order fill
+        /// </summary>
+        /// <param name="sender">For high resolution, sender object represents the data consolidator</param>
+        /// <param name="data">Last price trade bar available or consolidated</param>
+        /// <param name="orderId">Order fill ID</param>
+        /// <remarks>
+        /// The whole method body is locked, since order events can be triggered at any time in live mode
+        /// </remarks>
+        private void Update(object sender, IBaseData data, int orderId)
```

Comment:
Why pass `object sender` here? Since this method is not _directly_ used as an event handler, we shouldn't be using `object sender` but instead should be using something more strongly typed. Given the invocations above, it appears this is sometimes the transaction handler. It may make sense to define two separate methods, one for high resolution and one for low resolution since it looks like they'll be receiving different `sender` types at run time.